app/services/data_migrations/migrate_to_structured_offer_conditions.rb

```diff
@@ -0,0 +1,17 @@
+module DataMigrations
+  class MigrateToStructuredOfferConditions
+    TIMESTAMP = 20230504152212
+    MANUAL_RUN = false
+
+    def change
+      OfferCondition.where(type: nil).find_each do |unstructured_condition|
+        unstructured_condition.update!(
+          type: 'TextCondition',
+          details: {
+            description: unstructured_condition.text,   
+          },
+        )
+      end
```

Comment:
As a follow-up I think it makes sense to have a (regular) to remove the `text` column (as it will only contain duplicate data in future) and make the `type` column NOT NULL.

Reply:
agreed. make a card?

Reply:
https://trello.com/c/ADlyFkUv/1470-migration-to-cleanup-following-structured-conditions